Tests

fabian-hk opened the pull request:
Test environment with docker-compose and test cases defined as yaml files.

fabian-hk commented:
I'm not sure if we should rename the test case equals_operator.yml to domains_and_ips.yml because it basically tests the module domains_and_ips.py. What do you think about this?

fabian-hk commented:
I'm not sure exactly how the yaml files style guide works, but should we name the test cases with - instead of _?

danielfett commented:
> I'm not sure if we should rename the test case equals_operator.yml to domains_and_ips.yml because it basically tests the module domains_and_ips.py. What do you think about this?

Okay for me!

danielfett commented:
> I'm not sure exactly how the yaml files style guide works, but should we name the test cases with - instead of _?

I do not think that there is a consistent style. "-" is okay for me.

danielfett commented:
I get this when trying to run run.py:
```
danielf@www:~/git/yesses$ ./run.py --help
Traceback (most recent call last):
  File "./run.py", line 6, in <module>
    from yesses import Runner, all_modules, Config
  File "/home/danielf/git/yesses/yesses/__init__.py", line 1, in <module>
    from .config import Config
  File "/home/danielf/git/yesses/yesses/config.py", line 3, in <module>
    from .findingslist import FindingsList
  File "/home/danielf/git/yesses/yesses/findingslist.py", line 2, in <module>
    from .state import State
  File "/home/danielf/git/yesses/yesses/state.py", line 3, in <module>
    import yesses.scan.tls_settings
  File "/home/danielf/git/yesses/yesses/scan/__init__.py", line 5, in <module>
    from .information_leakage import InformationLeakage
  File "/home/danielf/git/yesses/yesses/scan/information_leakage.py", line 5, in <module>
    from comment_parser import comment_parser
ModuleNotFoundError: No module named 'comment_parser'
```
Did you check in all files?

danielfett commented on yesses/discover/error_paths.py:
```diff
@@ -0,0 +1,59 @@
+
+                with requests.Session() as req_sess:
+                    # get page with 404 not found error
+                    r = req_sess.get(f"{parsed_url.url_without_path}/fvwwvaoqgf/opdvsltqfnlcelh/ddsleo/glcgrfmr.odt",
```
I propose to change the URL to:
/yesses-scanner-nonexisting-url/opdvsltqfnlcelh/ddsleo/glcgrfmr.html

danielfett commented on yesses/discover/error_paths.py:
```diff
@@ -0,0 +1,59 @@
+
+class ErrorPaths(YModule):
+    """
+    Collect error pages to scan them for information leakages.
```
Please provide more information on what the module does.

danielfett commented on yesses/discover/hidden_paths.py:
```diff
@@ -0,0 +1,223 @@
+    """
+    This module takes urls and linked paths from the Linked Paths module.
+    It extracts potential folders from the linked paths and searches in
+    this folders with a wordlist for potential hidden files.
```
```suggestion
    these folders with a wordlist for potential hidden files.
```

danielfett commented on yesses/discover/hidden_paths.py:
```diff
@@ -0,0 +1,223 @@
+                parsed_url = utils.UrlParser(origin['url'])
+
+                # check if the web server replies to a random path which should not exist with a 200 status
+                r = requests.get(f"{parsed_url.url_without_path}/fvwwvaoqgf/opdvsltqfnlcelh/ddsleo/glcgrfmr.odt",
```
Similar URL to above.

danielfett commented on yesses/utils.py:
```diff
@@ -21,5 +27,138 @@ def patched_create_connection(address, *args, **kwargs):
+        parsed_url = UrlParser(origin['url'])
+        with force_ip_connection(origin['domain'], origin['ip']):
+            r = requests.get(parsed_url.url_without_path)
+            forwareded_parsed_url = UrlParser(r.url)
```
forwar**e**ded → forwarded

danielfett commented on yesses/utils.py:
```diff
@@ -21,5 +27,138 @@ def patched_create_connection(address, *args, **kwargs):
+            self.base_domain = '.'.join(split[1:])
+
+        # the url with the protocol and port (if no port is specified use a standard port)
+        self.url_without_path = self.parsed.netloc
```
Wouldn't that be the origin?

fabian-hk replied:
I have just googled it and according to this [website](https://developer.mozilla.org/en-US/docs/Web/API/URL/origin), you are right. I will rename it.

fabian-hk commented:
I think I have corrected all issues from the comments. I also moved the mini_config.yml file to docs/examples and renamed it to information_leakages_config.yml.
